server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregator.java

```diff
@@ -116,12 +130,13 @@ public InternalAggregation buildAggregation(long zeroBucket) throws IOException
 
         int num = Math.min(size, queue.size());
         final InternalComposite.InternalBucket[] buckets = new InternalComposite.InternalBucket[num];
-        int pos = 0;
-        for (int slot : queue.getSortedSlot()) {
+        int pos = queue.size() - 1;
+        while (queue.size() > 0) {
+            int slot = queue.pop();
             CompositeKey key = queue.toCompositeKey(slot);
             InternalAggregations aggs = bucketAggregations(slot);
             int docCount = queue.getDocCount(slot);
-            buckets[pos++] = new InternalComposite.InternalBucket(sourceNames, formats, key, reverseMuls, docCount, aggs);
+            buckets[pos--] = new InternalComposite.InternalBucket(sourceNames, formats, key, reverseMuls, docCount, aggs);
```

Comment:
we don't actually need to track a `pos` variable, we could use `queue.size()` instead?

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/36675/commits/b80398874b10bfce242373efe8b7f997ff610694